packages/interface/src/interface.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+interface PolyfillsWindow extends Window {
+  ShadyCSS?: ShadyCSS;
+}
+
+interface ShadyCSS {
+  nativeShadow: boolean;
+  nativeCss: boolean;
+
+  prepareTemplate(template: HTMLTemplateElement, elementName: string): void;
+  styleElement(element: HTMLElement, properties?: CustomPropertyValues): void;
+  styleSubtree(element: HTMLElement, properties?: CustomPropertyValues): void;
+  styleDocument(properties: CustomPropertyValues): void;
+
+  CustomStyleInterface?: CustomStyleInterface;
+}
+
+interface CustomStyleInterface {
+  addCustomStyle(style: HTMLStyleElement): void;
+}
+
+type CustomPropertyValues = { [property: string]: string };
+
+const polyfillsWindow = window as PolyfillsWindow;
```

Comment:
I think you can redeclare `window` rather than a new variable:

```ts
declare var window: PolyfillsWindow;
```

Reply:
Done.